source/common/res/features/toggle-splits/main.js

```diff
@@ -1,82 +1,42 @@
-      function setDisplayEnd() {
-        if (ynabToolKit.toggleSplits.setting === 'show') return;
-
-        calculatingDisplayEnd = true;
-        var rowView = ynabToolKit.shared.getEmberViewByContainerKey('view:ynab-grid/rows');
-        var accountsController = ynabToolKit.shared.containerLookup('controller:accounts');
-        var contentResults = accountsController.get('contentResults');
-        var containerView = rowView.get('containerView');
-        var gridView = rowView.get('gridView');
-        var displayStart = containerView.get('displayStart');
-        var recordsPerBody = containerView.get('recordsPerBody');
-
-        // this calculation was taken from YNAB's current codebase
-        var wouldBeDisplayEnd = Math.round(Math.min(displayStart + 4 * recordsPerBody, gridView.get('total')));
-
-        var fitOnPage = wouldBeDisplayEnd - displayStart;
-        var displayedCount = 0;
-        var displayEnd = displayStart;
-
-        for (var i = displayStart; i < contentResults.length; i++) {
-          if (displayedCount === fitOnPage) break;
-
-          if (contentResults[i].get('parentEntityId')) {
-            if (ynabToolKit.toggleSplits.setting === 'hide') {
-              displayEnd++;
-              continue;
-            } else {
-              displayedCount++;
+      // let grid = ynabToolKit.shared.getEmberView($('.ynab-gird').attr('id'));
+      let accountsController = ynabToolKit.shared.containerLookup('controller:accounts');
+      accountsController.reopen({
+        __toolkitShowSubTransactions: false,
+        content: Ember.computed({
+          get: function () { return []; },
+          set: function (key, val) {
+            if (!this.get('__toolkitShowSubTransactions')) {
+              val = val.filter((transaction) => {
+                let displayItemType = transaction.get('displayItemType');
+                return displayItemType !== ynab.constants.TransactionDisplayItemType.SubTransaction &&
+                       displayItemType !== ynab.constants.TransactionDisplayItemType.ScheduledSubTransaction;
+              });
             }
-          } else {
-            displayedCount++;
-            displayEnd++;
+            return val;
           }
-        }
-
-        Ember.run.next(function () {
-          containerView.set('displayEnd', displayEnd);
-          calculatingDisplayEnd = false;
-        });
-      }
-
-      function addScrollListener() {
-        var rowView = ynabToolKit.shared.getEmberViewByContainerKey('view:ynab-grid/rows');
-        rowView.get('containerView').addObserver('scrollTop', setDisplayEnd);
-
-        if (rowView.get('containerView').get('scrollTop') === 0) {
-          setDisplayEnd();
-        }
-      }
+        })
+      });
 
       function hideSubTransactions() {
         ynabToolKit.toggleSplits.setting = 'hide';
+        accountsController.set('__toolkitShowSubTransactions', false);
         $('.ynab-grid-body .ynab-grid-body-sub:not(.is-editing)').hide();
```

Comment:
Yeah you probably could change the `content` property in `component:register/grid-index` into a computed property instead. Currently all the transaction filtering code is in the accounts controller.

The route gets a raw list of transactions from the shared library and sets them as `visibleTransactionDisplayItems` on the accounts controller. The method `_filterContent` will take that array and sort/filter then set as `contentResults`. `contentResults` then gets set down on the components as just `content`.

So triggering `contentResults` will cause just the register to rerender vs triggering `visibleTransactionDisplayItems` or calling `_filterContent()` will do a sort/filter/rerender from the top.

Eventually all that code in the controller will be moved to the components as Ember will eventually remove controllers.